packages/i18nify-js/src/modules/phoneNumber/getDialCodeByCountryCode.ts

```diff
@@ -0,0 +1,29 @@
+import { withErrorBoundary } from '../../common/errorBoundary';
+import getDialCodes from './getDialCodes';
+
+/**
+ * Retrieves the dial code for a specified country code.
+ *
+ * @param countryCode The country code for which the dial code is to be retrieved.
+ *                    It must be a key of the object returned by the `getDialCodes` function.
+ * @returns The corresponding dial code as a string.
+ * @throws An error if the provided country code is not found in the dial code mapping.
+ */
+const getDialCodeByCountryCode = (
+  countryCode: keyof ReturnType<typeof getDialCodes>,
```

Comment:
create a type for this and export it for consumers as well. 
Check this [type](https://github.com/razorpay/i18nify/blob/efd1f645454517789111c8d9ba8582e86e8a9a62/packages/i18nify-js/src/modules/currency/types.ts#L59) for reference